src/Sentry/Internal/BackgroundWorker.cs

```diff
@@ -59,20 +69,37 @@ public bool EnqueueEnvelope(Envelope envelope)
 
             if (Interlocked.Increment(ref _currentItems) > _maxItems)
             {
-                _ = Interlocked.Decrement(ref _currentItems);
+                Interlocked.Decrement(ref _currentItems);
                 _options.ClientReportRecorder.RecordDiscardedEvents(DiscardReason.QueueOverflow, envelope);
+                _options.LogDebug("Discarding envelope {0} because the queue is full.", envelope.TryGetEventId());
```

Comment:
Worthy of "info" level?

```suggestion
                _options.LogInfo("Discarding envelope {0} because the queue is full.", envelope.TryGetEventId());
```